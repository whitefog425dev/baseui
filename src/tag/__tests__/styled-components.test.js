/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {StyledRoot, StyledAction} from '../index.js';

describe('Tag styled components', () => {
  describe('StyledRoot', () => {
    test.each([[''], ['$disabled'], ['$isFocused'], ['$isActive']])(
      '',
      prop => {
        const props = {};
        props[prop] = true;
        const component = shallow(
          <StyledRoot {...props}>
            <div />
          </StyledRoot>,
        );
        expect(component.instance().getStyles()).toMatchSnapshot(
          'StyledRoot has correct styles when ' + prop,
        );
      },
    );
  });

  describe('StyledAction', () => {
    test.each([['$disabled'], ['']])('', prop => {
      const props = {};
      props[prop] = true;
      const component = shallow(
        <StyledAction {...props}>
          <div />
        </StyledAction>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledAction has correct styles when ' + prop,
      );
    });
  });
});
