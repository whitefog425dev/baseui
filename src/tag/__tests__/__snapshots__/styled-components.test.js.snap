// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag styled components StyledAction : StyledAction has correct styles when  1`] = `
Object {
  ":hover": Object {
    "backgroundColor": null,
  },
  "alignItems": "center",
  "borderBottomRightRadius": "$theme.borders.radius400",
  "borderTopRightRadius": "$theme.borders.radius400",
  "cursor": "pointer",
  "display": "flex",
  "marginLeft": "8px",
  "paddingBottom": "8px",
  "paddingLeft": "8px",
  "paddingRight": "8px",
  "paddingTop": "8px",
}
`;

exports[`Tag styled components StyledAction : StyledAction has correct styles when $disabled 1`] = `
Object {
  ":hover": Object {
    "backgroundColor": null,
  },
  "alignItems": "center",
  "borderBottomRightRadius": "$theme.borders.radius400",
  "borderTopRightRadius": "$theme.borders.radius400",
  "cursor": "not-allowed",
  "display": "flex",
  "marginLeft": "8px",
  "paddingBottom": "8px",
  "paddingLeft": "8px",
  "paddingRight": "8px",
  "paddingTop": "8px",
}
`;

exports[`Tag styled components StyledRoot : StyledRoot has correct styles when  1`] = `
Object {
  ":hover": Object {
    "backgroundColor": null,
  },
  "alignItems": "center",
  "backgroundColor": null,
  "borderColor": undefined,
  "borderRadius": "$theme.borders.radius400",
  "borderWidth": "1px",
  "boxSizing": "border-box",
  "color": undefined,
  "cursor": "auto",
  "display": "inline-flex",
  "fontFamily": "$theme.typography.font250.fontFamily",
  "fontSize": "$theme.typography.font250.fontSize",
  "fontWeight": "$theme.typography.font250.fontWeight",
  "height": "$theme.sizing.scale800",
  "justifyContent": "space-between",
  "lineHeight": "$theme.typography.font250.lineHeight",
  "marginBottom": "5px",
  "marginLeft": "5px",
  "marginRight": "5px",
  "marginTop": "5px",
  "opacity": null,
  "paddingBottom": "$theme.sizing.scale100",
  "paddingLeft": "$theme.sizing.scale500",
  "paddingTop": "$theme.sizing.scale100",
}
`;

exports[`Tag styled components StyledRoot : StyledRoot has correct styles when $disabled 1`] = `
Object {
  ":hover": Object {
    "backgroundColor": null,
  },
  "alignItems": "center",
  "backgroundColor": null,
  "borderColor": undefined,
  "borderRadius": "$theme.borders.radius400",
  "borderWidth": "1px",
  "boxSizing": "border-box",
  "color": undefined,
  "cursor": "not-allowed",
  "display": "inline-flex",
  "fontFamily": "$theme.typography.font250.fontFamily",
  "fontSize": "$theme.typography.font250.fontSize",
  "fontWeight": "$theme.typography.font250.fontWeight",
  "height": "$theme.sizing.scale800",
  "justifyContent": "space-between",
  "lineHeight": "$theme.typography.font250.lineHeight",
  "marginBottom": "5px",
  "marginLeft": "5px",
  "marginRight": "5px",
  "marginTop": "5px",
  "opacity": ".56",
  "paddingBottom": "$theme.sizing.scale100",
  "paddingLeft": "$theme.sizing.scale500",
  "paddingTop": "$theme.sizing.scale100",
}
`;

exports[`Tag styled components StyledRoot : StyledRoot has correct styles when $isActive 1`] = `
Object {
  ":hover": Object {
    "backgroundColor": null,
  },
  "alignItems": "center",
  "backgroundColor": null,
  "borderColor": undefined,
  "borderRadius": "$theme.borders.radius400",
  "borderWidth": "1px",
  "boxSizing": "border-box",
  "color": undefined,
  "cursor": "auto",
  "display": "inline-flex",
  "fontFamily": "$theme.typography.font250.fontFamily",
  "fontSize": "$theme.typography.font250.fontSize",
  "fontWeight": "$theme.typography.font250.fontWeight",
  "height": "$theme.sizing.scale800",
  "justifyContent": "space-between",
  "lineHeight": "$theme.typography.font250.lineHeight",
  "marginBottom": "5px",
  "marginLeft": "5px",
  "marginRight": "5px",
  "marginTop": "5px",
  "opacity": null,
  "paddingBottom": "$theme.sizing.scale100",
  "paddingLeft": "$theme.sizing.scale500",
  "paddingTop": "$theme.sizing.scale100",
}
`;

exports[`Tag styled components StyledRoot : StyledRoot has correct styles when $isFocused 1`] = `
Object {
  ":hover": Object {
    "backgroundColor": null,
  },
  "alignItems": "center",
  "backgroundColor": null,
  "borderColor": undefined,
  "borderRadius": "$theme.borders.radius400",
  "borderWidth": "1px",
  "boxSizing": "border-box",
  "color": undefined,
  "cursor": "auto",
  "display": "inline-flex",
  "fontFamily": "$theme.typography.font250.fontFamily",
  "fontSize": "$theme.typography.font250.fontSize",
  "fontWeight": "$theme.typography.font250.fontWeight",
  "height": "$theme.sizing.scale800",
  "justifyContent": "space-between",
  "lineHeight": "$theme.typography.font250.lineHeight",
  "marginBottom": "5px",
  "marginLeft": "5px",
  "marginRight": "5px",
  "marginTop": "5px",
  "opacity": null,
  "paddingBottom": "$theme.sizing.scale100",
  "paddingLeft": "$theme.sizing.scale500",
  "paddingTop": "$theme.sizing.scale100",
}
`;
